### DISCUSSION TITLE
How to downsample the number of points in each cell

### DISCUSSION THREAD

**Discussion by zixiao-yin** [pyvista/pyvista#6144] [Q&A] [answered]
### First check

- [X] I added a very descriptive title here.
- [x] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the PyVista documentation, with the integrated search.
- [X] I already searched in Google "How to X in PyVista" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I will NOT make duplicate discussions or issues in the future.

### Commit to Help

- [X] I commit to help with one of those options.

### Sample Code What is the problem, question, or error?
Write a short description telling me what you are doing, what you expect to happen, and what is currently happening.

```python
mesh = pv.read(r'F:\0_lead_connectomes\PPMI_100%_fiber_mesh.vtk')
```

### Description

Hi dear community, I have a fiber tract mesh (all of them are simple lines) with 632428 cells and 28197120 points in total (which means around 44 points for each cell). I want to reduce the number/resolution of points in each cell, e.g., making it an average a 20 points for each cell. However, I didn't find anyway to do that through pyvista. Could you please let me know if this is doable? Thanks a lot in advance!

### System Information

```shell
Date: Wed May 29 12:15:28 2024 西欧夏令时

                OS : Windows (10 10.0.19045 SP0 Multiprocessor Free)
            CPU(s) : 8
           Machine : AMD64
      Architecture : 64bit
               RAM : 31.9 GiB
       Environment : Jupyter
        GPU Vendor : Intel
      GPU Renderer : Intel(R) UHD Graphics 620
       GPU Version : 4.5.0 - Build 30.0.101.1122
  MathText Support : False

  Python 3.10.4 | packaged by conda-forge | (main, Mar 24 2022, 17:32:50) [MSC
  v.1929 64 bit (AMD64)]

           pyvista : 0.43.8
               vtk : 9.3.0
             numpy : 1.22.3
        matplotlib : 3.8.2
            scooby : 0.10.0
             pooch : 1.8.0
            pillow : 9.1.0
           imageio : 2.34.0
           IPython : 8.3.0
        ipywidgets : 8.1.1
             scipy : 1.8.0
              tqdm : 4.66.1
        jupyterlab : 4.1.4
      nest_asyncio : 1.5.5
```

### Screenshots

![微信截图_20240529121600](https://github.com/pyvista/pyvista/assets/55424621/94c1bba5-78ad-4175-aa2f-9f9e71bc8222)

**Comment by MatthewFlamm** [comment, marked as answer]
vtk has [vtkDecimatePolylineFilter](https://vtk.org/doc/nightly/html/classvtkDecimatePolylineFilter.html#details), but this isn't included in PyVista yet.  This would be a nice feature add.

**Reply by zixiao-yin** [reply]
Wow! This is so cool. I made it with tkDecimatePolylineFilter. Thank you soooo much!!

**Reply by zixiao-yin** [reply]
Below are the codes in case anyone also needs to do this:

```python
# Create the decimation filter
decimate = vtk.vtkDecimatePolylineFilter()
decimate.SetInputData(mesh) # This is my pyvista mesh
decimate.SetTargetReduction(0.5)  # 0.5 means reduce 50% of points

# Update the filter
decimate.Update()

# Get the output (decimated) polydata
decimated_polydata = decimate.GetOutput()

# Print the number of points after decimation
decimated_points = decimated_polydata.GetPoints()
print(f"Number of points after decimation: {decimated_points.GetNumberOfPoints()}")

# Transform the polydata back to pyvista format
mesh_decimated = pv.PolyData(decimated_polydata) 
```

**Reply by MatthewFlamm** [reply]
Implemented in https://github.com/pyvista/pyvista/pull/6153
